src/os/ui/layer/ellipsecolumns.js

```diff
@@ -0,0 +1,356 @@
+    const layerId = instanceOf(layer, os.layer.Vector.NAME) ? layer.getId() : undefined;
+    const descMappings = layerId ?
+      (DataManager.getInstance().getDescriptor(layerId).getMappings() || []) : layer['mappings'];
+    const mappings = this.createMappings();
+
+    let result = [...descMappings];
+
+    // Only combine the mappings if passed a layer, don't if passed anything else (used by geometrystep)
+    if (layerId && mappings.length != 0 && descMappings.length != 0) {
+      mappings.forEach((mapping) => {
+        const im = result.findIndex(({toField}) => toField === mapping.toField);
+        if (im >= 0) {
+          result[im] = mapping;
+        } else {
+          result.push(mapping);
+        }
+      });
+    } else if (descMappings.length == 0) {
+      result = mappings;
+    }
+
+    this.scope_.$parent['confirmValue'] = layerId ? result : mappings;
+  }
+
+  /**
+   * Create Mappings for Ellipse Data
+   * @return {Array<AbstractMapping>}
+   */
+  createMappings() {
+    const mappings = [];
+    const type = this['inputType'];
+
+    if (type == 0 && this['radiusColumn'] && this['radiusUnits']) {
+      const rm = new RadiusMapping();
+      rm.field = this['radiusColumn'].name;
+      rm.setUnits(this['radiusUnits']);
+      mappings.push(rm);
+    }
+    if (type == 1 && this['semiMajorColumn'] && this['semiMajorUnits']) {
+      var smaj = new SemiMajorMapping();
+      smaj.field = this['semiMajorColumn'].name;
+      smaj.setUnits(this['semiMajorUnits']);
+      mappings.push(smaj);
+    }
+    if (type == 1 && this['semiMinorColumn'] && this['semiMinorUnits']) {
+      var smin = new SemiMinorMapping();
+      smin.field = this['semiMinorColumn'].name;
+      smin.setUnits(this['semiMinorUnits']);
+      mappings.push(smin);
+    }
+    if (type == 1 && this['orientation']) {
+      var om = new OrientationMapping();
+      om.field = this['orientation'].name;
+      mappings.push(om);
+    }
+
+    return mappings;
+  }
+}
+
+
+/**
+ * Settings key for if this capability is enabled in configs
+ * @type {string}
+ */
+const ALLOW_ELLIPSE_CONFIG = 'allowEllipseConfiguration';
+
+
+/**
+ * Launches the window to configure ellipse columns
+ * @param {*} layer
+ * @param {function(Array<AbstractMapping>)=} opt_confirmCallback
+ */
+const launchConfigureWindow = function(layer, opt_confirmCallback) {
+  const confirm = opt_confirmCallback || callback_.bind(this, layer);
+  const scopeOptions = {
+    'layer': layer
+  };
+
+  const options = /** @type {osx.window.ConfirmOptions} */ ({
+    confirm: confirm,
+    prompt: '<ellipsecolumns layer="layer"></ellipsecolumns>',
+    yesText: 'Apply and Reload',
+    yesButtonClass: 'btn-danger',
+    windowOptions: {
+      'label': 'Map Ellipse Columns',
+      'x': 'center',
+      'y': 'center',
+      'width': '330',
+      'height': 'auto',
+      'modal': 'true',
+      'show-close': 'true',
+      'headerClass': 'bg-danger u-bg-danger-text'
```

Comment:
The red header makes it feel like I'm doing something inherently dangerous/risky, which I don't think is appropriate for this window. I would personally vote for using the standard header/button for this UI.